File: ruby/lib/enolib/elements/missing/missing_fieldset.rb
# frozen_string_literal: true

module Enolib
  class MissingFieldset < MissingElementBase
    def entries(_key = nil)
      []
    end

    def entry(key = nil)
      MissingFieldsetEntry.new(key, self)
    end

    def optional_entry(_key = nil)
      nil
    end

    def required_entry(_key = nil)
      @parent._missing_error(self)
    end

    def to_s
      if @key
        "#<Enolib::MissingFieldset key=#{@key}>"
      else
        '#<Enolib::MissingFieldset>'
      end
    end
  end
end

File: python/enolib/elements/missing/missing_fieldset.py
from .missing_element_base import MissingElementBase
from .missing_fieldset_entry import MissingFieldsetEntry

class MissingFieldset(MissingElementBase):
    def __repr__(self):
        if self._key:
            return f"<class MissingFieldset key={self._key}>"

        return '<class MissingFieldset>'

    def entries(self, _key=None):
        return []

    def entry(self, key=None):
        return MissingFieldsetEntry(key, self)

    def optional_entry(self, _key=None):
        return None

    def required_entry(self, _key=None):
        self._parent._missing_error(self)
